### PR TITLE
EXT-986 Fixed backward compatibility of input files without domains.

### PR DESCRIPTION
**Pull request by KevinMitchellREDBRICK**
Fixed backward compatibility of input files without domains. Removed requirement that input files be named '.json'

### CODE REVIEW

File: src/js/popup.js [hunk 1]
```diff
@@ -469,25 +469,34 @@ $(document).ready(function () {
...
-                    }, function (response) {
-                        if (response.status === 'success') {
-                            location.reload();
-                            Materialize.toast('Templates successfully loaded from file', 2000, 'toastNotification');
-                            dmTemplateUpdate('upload');
-                        } else {
-                            if (response.message) {
-                                dmError('upload', response.message);
-                                Materialize.toast(response.message, 2000, 'toastNotification');
+                    let validJSON = true;
```
Comment (cfoucher) [inline src/js/popup.js]: We're already parsing the JSON in the background page. This try/catch isn't necessary.

File: src/js/popup.js [hunk 2]
```diff
@@ -469,25 +469,34 @@ $(document).ready(function () {
...
+                            if (response.status === 'success') {
+                                location.reload();
+                                Materialize.toast('Templates successfully loaded from file', 2000, 'toastNotification');
+                                dmTemplateUpdate('upload');
                             } else {
-                                dmError('upload', 'generic');
-                                Materialize.toast('Something went wrong. Please try again.', 2000, 'toastNotification');
+                                if (response.message) {
+                                    dmError('upload', response.message);
+                                    Materialize.toast(response.message, 2000, 'toastNotification');
```
Comment (cfoucher) [inline src/js/popup.js]: I'm assuming https://github.com/rdbrck/jira-template-injector/pull/41/files#diff-9566d9c6e04bce6d87d5cbfbf1062b04L449 erroring was the reason nothing was happening for both the upload and download buttons and the reason this wasn't showing anything?

File: src/js/popup.js (hunk 1 above)
Comment (KevinMitchellREDBRICK) [reply src/js/popup.js]: :+1:

File: src/js/popup.js (hunk 2 above)
Comment (KevinMitchellREDBRICK) [reply src/js/popup.js]: Yes. I wasn't checking that domains existed. Now fixed.

**Pull request merged**